### PR TITLE
Minor clean-up to SqlResult

### PR DESCRIPTION
**Pull request by pauldraper**
Part of this answers a problem @krjackso had.

There's no reason for `BigInteger` stuff.

### CODE REVIEW

**Comment by krjackso** [comment]
Running this locally I had two tests fail:

[info] javaBigDecimal should
[info] x return the correct value
[error]    'Some(1.0)' is Some but '1.0' is not equal to '1' (SqlResultSpec.scala:741)

[info] extractOption should
[info] + Extract a Some
[info] x Extract a None
[error]    'Some(hello)' is not None (SqlResultSpec.scala:200)

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 1]
```diff
@@ -195,15 +198,15 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
   def strictURLOption(column: String): Option[URL] = getResultSetOption(resultSet.getURL(column))
 
   protected[relate] def extractOption[A](column: String)(f: (Any) => A): Option[A] = {
-    resultSet.getObject(column).asInstanceOf[Any] match {
-      case x if (x == null || resultSet.wasNull()) => None
+    resultSet.getObject(column) match {
+      case x if resultSet.wasNull => None
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: Why did you remove the check (x == null) ?

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 2]
```diff
@@ -195,15 +198,15 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
...
       case x => Some(f(x))
     }
   }
 
-  protected def getResultSetOption[A](f: => A): Option[A] = {
+  protected def getResultSetOption[A](f: A): Option[A] = {
     f match {
-      case x if (x == null || resultSet.wasNull()) => None
+      case x if resultSet.wasNull => None
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: Why did you remove the check (x == null) ?

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 3]
```diff
@@ -239,13 +242,15 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
     }
   }
 
+  @deprecated("Use long(String) or string(String) instead", "1.8.0")
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: mysql requires scala/java bigint for BIGINT UNSIGNED. Because of this, mysql will use bigints for ALL autoincrement columns. We can't deprecate this for longs.

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 4]
```diff
@@ -239,13 +242,15 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
...
   def bigInt(column: String): BigInt = bigIntOption(column).get
+  @deprecated("Use longOption(String) or stringOption(String) instead", "1.8.0")
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: can't deprecate this either, same reason.

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 5]
```diff
@@ -255,11 +260,13 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
       case x: Int => BigDecimal(x)
       case x: Long => BigDecimal(x)
       case x: String => BigDecimal(x)
-      case x: java.math.BigDecimal => BigDecimal(x.toString)
+      case x: java.math.BigDecimal => BigDecimal(x)
     }
   }
 
+  @deprecated("Use long(String) or string(String) instead", "1.8.0")
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: can't deprecate this either, same reason.

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala [hunk 6]
```diff
@@ -255,11 +260,13 @@ class SqlResult(val resultSet: java.sql.ResultSet) {
...
   def javaBigInteger(column: String): java.math.BigInteger = javaBigIntegerOption(column).get
+  @deprecated("Use longOption(String) or stringOption(String) instead", "1.8.0")
```
Comment (matthew-lucidchart) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: can't deprecate this either, same reason.

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala (hunk 1 above)
Comment (pauldraper) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: It's unnecessary. `resultSet.wasNull` is already checks that.

**Comment by matthew-lucidchart** [comment]
This looks like the kind of commit where you didn't know why things were written the way they were, and it looked dumb, so you got rid of them. Was there a problem you were trying to solve? If not, I'm going to reject, cause it will break us.

File: src/main/scala/com/lucidchart/open/relate/SqlResult.scala (hunk 1 above)
Comment (pauldraper) [inline src/main/scala/com/lucidchart/open/relate/SqlResult.scala]: They're equivalent. Though actually, it's probably best (most performant) to remove the other one instead.

> If the value is an SQL NULL, the driver returns a Java null

**Comment by pauldraper** [comment]
Keilan had the problem where he could get a `BigInteger`, but he couldn't set one (via interpolation).
Presumably, we should support the same data types both ways.

Options are to (1) remove the get method, (2) add a set method.

(2) is possible, though the JDBC interface doesn't support it. We could convert the `BigInteger` to something that is support, like `BigDecimal` or `String`, and hopefully the right thing would happen, but there's no guarantee, and it's all implementation-dependent.

But it sounds like (2) is what we would prefer. Probably with `String` conversion, the universal data type.

**Comment by matthew-lucidchart** [comment]
Yes, I think having a set method is valuable.

**Comment by krjackso** [comment]
Since adding a set method is completely different than the work that was done in this branch, I'm going to close the pull request.  If/when we add a set method we can create another one, but I would probably recommend @matthew-lucidchart  to review it over me, since I don't have as good an understanding of how changes to relate would affect us, or about mysql data types

**Comment by pauldraper** [comment]
Except for the other stuff...
